pkg/network/ebpf/c/tracer.c

```diff
@@ -370,7 +387,10 @@ int kprobe__ip6_make_skb(struct pt_regs *ctx) {
     // commit: https://github.com/torvalds/linux/commit/f37a4cc6bb0ba08c2d9fd7d18a1da87161cbb7f9
     struct inet_cork_full *cork_full = (struct inet_cork_full *)PT_REGS_PARM9(ctx);
     struct flowi6 *fl6 = &cork_full->fl.u.ip6;
-#elif !defined(COMPILE_RUNTIME) || LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0)
+#elif defined(COMPILE_CORE)
+    struct inet_cork_full *cork_full = (struct inet_cork_full *)PT_REGS_PARM9(ctx);
+    struct flowi6 *fl6 = (struct flowi6 *)__builtin_preserve_access_index(&cork_full->fl.u.ip6);
```

Comment:
`BPF_CORE_READ` won't work here?

Reply:
Would just using `bpf_core_read` here work? i.e., not use the macro. Would be better to hide that `__builtin_preserve_access_index` bit.

Reply:
That still does a `bpf_probe_read_kernel` under the hood, which is what we don't want.